### PR TITLE
Feature http service

### PR DESCRIPTION
**Pull request by blackbaud-brandonhare**
Version 2

### CODE REVIEW

File: src/app/app-extras.module.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import { NgModule } from '@angular/core';
-
```
Comment (Blackbaud-BrandonJones) [inline src/app/app-extras.module.ts]: I think we want this blank line back.  Separates our local imports from the node_modules.

File: src/app/public/src/modules/shared/search-result.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    title: string;
+    site_name: string;
+    host: string;
+    path: string;
+    text: string;
+  };
+  highlights?: {
+    text: string[]
+  };
+
```
Comment (Blackbaud-BrandonJones) [inline src/app/public/src/modules/shared/search-result.ts]: extra blank line?

File: src/app/public/src/modules/shared/search-results-provider.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  private siteNames: string[] = [];
+  private endpoints: any = {
+    internal: 'query',
+    public: 'query-public'
+  };
+  private queryUrlBase: string = 'https://stache-search-query.sky.blackbaud.com';
+
+  constructor(
+    private http: StacheHttp,
+    private config: StacheConfigService
```
Comment (Blackbaud-BrandonJones) [inline src/app/public/src/modules/shared/search-results-provider.ts]: I feel like we can eliminate a ton of the verbose lines of code.   I see only one place where we are actually using `this.config._____` without going all the way down into `config.skyux.appSettings.stache.searchConfig`.  Couldn't we just create a method like

```
validateSearchConfig(): any {
  let searchConfig = {};

  if (this.configService.skyux.appsettings 
    && this.configService.skyux.appSettings.stache
    && this.configService.skyux.appSettings.stache.searchConfig)  {
    searchConfig = this.configService.skyux.appSettings.stache.searchConfig;
  }

  this.searchConfig = searchConfig;
}
```

Comment (Blackbaud-SteveBrush) [reply src/app/public/src/modules/shared/search-results-provider.ts]: Valid point.

**Pull request closed without merging**